From f8b0e03f498d6f9e377a9ca22ec032c8061e5530 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Thu, 30 Jan 2014 00:51:42 +0100
Subject: [PATCH 86/87] MEDIUM: http: make keep-alive + httpclose be passive
 mode

There's no particular reason for having keep-alive + httpclose combine
into forceclose when set in different frontend/backend sections, since
keep-alive does not close anything by default. Let's have this still
combination remain httpclose only.
---
 src/proto_http.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/proto_http.c b/src/proto_http.c
index 164faac..6d60fad 100644
--- a/src/proto_http.c
+++ b/src/proto_http.c
@@ -3568,8 +3568,8 @@ int http_process_req_common(struct session *s, struct channel *req, int an_bit,
 		    (s->be->options & PR_O_HTTP_MODE) == PR_O_HTTP_FCL)
 			tmp = TX_CON_WANT_CLO;
 
-		/* option httpclose + anything other than tunnel => close */
-		if (tmp != TX_CON_WANT_TUN &&
+		/* option httpclose + server_close => forceclose */
+		if (tmp == TX_CON_WANT_SCL &&
 		    ((s->fe->options & PR_O_HTTP_MODE) == PR_O_HTTP_PCL ||
 		     (s->be->options & PR_O_HTTP_MODE) == PR_O_HTTP_PCL))
 			tmp = TX_CON_WANT_CLO;
-- 
1.8.3.2

